design/dedicated-osd-pod.md

```diff
@@ -0,0 +1,147 @@
+- Tight Ceph coupling. The monolithic device discovery and provisioning code cannot be reused for other backends.
+- Process management issue. Rook's process management is very simple. Using Kubernetes pod management is much more reliable.
+
+
+A more comprehensive discussions can be found at [this issue](https://github.com/rook/rook/issues/1341).
+
+## Proposal   
+
+We propose the following change to address the limitations.
+
+### Terms
+
+- Device Discovery. A Pod that is given a device or directory path filter and use the filter to discover devices on the host. The Pod updates the device orchestration Configmap with the device paths and mark the orchestration status as "DeviceDiscovered". The Pod is created by Rook Operator and runs in a Kubernetes deployment on the storage node. This process is storage backend agnostic. 
+
+- Device Provisioner. A Pod that is given device or directory paths upon start and make backend specific storage types. For instance, the provisioner prepares OSDs for Ceph backend. It is a Kubernetes batch job and exits after the devices are prepared.
+
+### Create new OSDs
+| Sequence |Rook Operator | Device Discovery  | Device Provisioner   | Ceph OSD Deployment | 
+|---|---|---|---|---|
+| 1  | Read devices from cluster CRD and create an OSD Discovery deployment on each storage node  |   |   |
+| 2  | Watch raw device Configmap  | Discover devices that fit those defined in device filters and update raw device Configmap with the device paths and filtering results| | |
+| 3  | Detect raw device Configmap change, parse Configmap, extract device paths with matching filtering results, and create an Device Provisioner deployment for each device  || | |
+| 4  | Watch device provisioning Configmap | | Prepare OSDs, Persist OSD ID, datapath, and node info in a per node device provisioning Configmap | |
+| 5  | Detect device provisioning Configmap change, parse Configmap, extract OSD info, construct OSD Pod command and arg | | | |
+| 6  | Create one deployment per OSD | | |
+| 7  | | | | Start `ceph-osd` Daemon one Pod per device |
+
+This change addresses the above limitations in the follow ways:
+- High reliability. Each `ceph-osd` daemon runs its own Pod, their restart and upgrade are by Kubernetes controllers. Upgrading Device Provisioner Pod no longer restarts `ceph-osd` daemons.
+- More efficient resource requests. Once Device Discovery detects all devices, Rook Operator is informed of the topology and assigns appropriate resources to each Ceph OSD deployment.
+- Reusable. Device discovery can be used for other storage backends.
+
+
+### Detailed Device Discovery Process
+
+Each `Device Discovery` Pod reads device filters from `Device Filter Configmap`, retrieves named device filters, walks through device trees and uses device filters to identify the devices, and stores the filtering results in `Raw Device Configmap`. 
```

Comment:
what is in the device filter? The simple place to start would be our current filter, which includes all raw devices that are not formatted.